composer.json

```diff
@@ -3,16 +3,65 @@
   "description": "Apache Avro™ is a data serialization system.",
   "minimum-stability": "stable",
   "license": "Apache-2.0",
+  "homepage": "http://avro.apache.org",
+  "type": "library",
+  "keywords": [
+    "RPC"
+  ],
+  "readme": "README.md",
+  "authors": [
+    {
+      "name": "Apache Avro Developers",
+      "email": "dev@avro.apache.org",
+      "homepage": "http://avro.apache.org"
+    }
+  ],
+  "support": {
+    "email": "dev@avro.apache.org",
+    "issues": "https://issues.apache.org/jira/browse/AVRO"
+  },
   "require": {
-    "beberlei/composer-monorepo-plugin": "0.16.5"
```

Comment:
https://issues.apache.org/jira/browse/AVRO-2752?focusedCommentId=17870165&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17870165
According to Siad this plugin does something important here! But AFAIU the idea is to use `archive > exclude` below instead!

Reply:
My packagist-fu is almost non-existent, but Thrift also uses a monorepo-based setup, and in their case they don't seem to need that plugin. I'm more than happy to re-instate the plugin if it is actually needed. I'll try to take a look at those failing tests

Reply:
My understanding here is that the monorepo plugin is designed to allow publishing multiple php packages from the same repository. The important part it's currently doing is using the [monorepo.json](https://github.com/apache/avro/blob/35b01df443445e5a9193b335b0416075ec469b3a/lang/php/monorepo.json) file as a discovery-mechanism of sorts to locate the php source files. 

Unless I'm missing something the `autoload.psr-4` section of this new version will serve that need instead via [PSR-4 autoloading](https://getcomposer.org/doc/04-schema.md#psr-4) and the `archive.exclude` with the exclamation mark ensures only the php parts of this repository are included.

I think we could even delete the `monorepo.json` file too.

Reply:
I also currently hit install issues with the old version. I'm testing with a brand new empty project, with the following `composer.json`:

```json
{
    "require": {
        "apache/avro": "dev-main"
    },
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/apache/avro"
        }
    ],
    "config": {
        "allow-plugins": {
            "beberlei/composer-monorepo-plugin": true
        }
    }
}
```

```bash
composer install
No composer.lock file present. Updating dependencies to latest instead of installing from lock file. See https://getcomposer.org/install for more information.
Loading composer repositories with package information
Updating dependencies
Lock file operations: 2 installs, 0 updates, 0 removals
  - Locking apache/avro (dev-main 35b01df)
  - Locking beberlei/composer-monorepo-plugin (v0.16.5)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 1 install, 1 update, 0 removals
  - Installing beberlei/composer-monorepo-plugin (v0.16.5): Extracting archive
  - Upgrading apache/avro (dev-AVRO-2843-packagist 4c1ec59 => dev-main 35b01df): Extracting archive
Generating autoload files
Generating autoload files for monorepo sub-packages with dev-dependencies.
PHP Fatal error:  Declaration of Monorepo\Composer\EventDispatcher::dispatch($eventName, ?Composer\EventDispatcher\Event $event = null) must be compatible with Composer\EventDispatcher\EventDispatcher::dispatch(?string $eventName, ?Composer\EventDispatcher\Event $event = null): int in /Users/aquaile/Development/avro-tests/vendor/beberlei/composer-monorepo-plugin/src/main/Monorepo/Composer/EventDispatcher.php on line 9

Fatal error: Declaration of Monorepo\Composer\EventDispatcher::dispatch($eventName, ?Composer\EventDispatcher\Event $event = null) must be compatible with Composer\EventDispatcher\EventDispatcher::dispatch(?string $eventName, ?Composer\EventDispatcher\Event $event = null): int in /Users/aquaile/Development/avro-tests/vendor/beberlei/composer-monorepo-plugin/src/main/Monorepo/Composer/EventDispatcher.php on line 9
```

Pointing to the new branch with this works:

```json
{
    "require": {
        "apache/avro": "dev-AVRO-2843-packagist"
    },
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/cv-library/avro"
        }
    ]
}

```